### DISCUSSION TITLE
Set Parameter Value through Lua

### DISCUSSION THREAD

**Discussion by TijeOortwijnThales** [helgoboss/helgobox#1334] [Q&A]
I got Realearn setup nicely with my Softube Console 1 and Stream Deck.

My Console 1 is used to scroll / browse through tracks and select them. Been using it the past few weeks.

My Stream Deck displays all the track names and colors + it highlights the currently selected track (changing the background color to white). On pushing the button, i display basic track controls for the selected tracks + the first 5 FX inserts for easy floating FX's and controlling them with my console 1. All working great. See the attached video.

The last thing I want to do is automatically page the streamdeck when I browse to a track beyond the ones currently displayed. Let the streamdeck follow the selection. I have a paging parameter which I can manually increment / decrement using a button on the stream deck, but I would love to set it through lua when browsing track with the console1. There's a couple of ways to do it  I guess, but I can't get it to work:

My main problem is:
One Instance (instance1) of Realearn is used to map the console 1. The important mapping here is Console 1 Input Gain button is mapped to Project -> Browse Track

Seconds instance (instance2) is mapping the stream deck. It has a parameter called 'Paging' 

I would like instance1 of realearn to update the parameter 'paging' on instance2. How would I do that?

I think setting up a compartment wide script that keeps track of the currently selected track and updates the paging parameter accordingly might do the trick too. I know that the possibility of multiple tracks being selected makes it more complex. I can't seem to find a good reference on how to set something like this up. 

 

https://github.com/user-attachments/assets/bfd31ffe-5dca-43ce-97d5-201f6da4c86a

**Comment by helgoboss** [comment]
The compartment parameters in the main unit of each instance are exposed as normal VST FX parameters. So you can use target "FX parameter: Set value" from anywhere to change the compartment parameters of another instance.
